### PR TITLE
Query and update

### PR DESCRIPTION
**Pull request by TesheMaximillan**
> In the second milestone of this project, I used transactions to update and delete records. Also, I used aggregate functions and GROUP BY to answer analytical questions.

# Basic Requirements Milestone II:

* A species column is added with a type of string to the animals' table.
* I inserted a data into the animals' table
* Inside a transaction I updated the animals' table by setting the species column to unspecified. 
Also, I Verified that the change was made. Then I rolled back the change and verify that species columns went back to the state before the transaction.
* Inside a transaction, I updated and Commit the transaction.
* Inside the transaction I deleted all records in the animal table, then I rolled back the transaction;
* Inside the transaction I used a Savepoint for rollback;
* Finally, I wrote a query to answer the listed question in the requirement.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Gabumon','2018-11-15',2,TRUE,8);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu','2021-01-07',1,FALSE,15.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon','2017-05-12',5,TRUE,11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander','2020-08-02',0,FALSE,11);
```
Comment (arslanbisharat) [inline data.sql]: - [ ] Please check the `weight_kg` is -11

File: data.sql [hunk 2]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon','2021-11-15',2,TRUE,5.7);
```
Comment (arslanbisharat) [inline data.sql]: - [ ] Please check the `weight_kg` is -5.7

File: data.sql [hunk 3]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle','1993-04-02',3,FALSE,12.13);
```
Comment (arslanbisharat) [inline data.sql]: - [ ] Please check the `weight_kg` is -12.13

File: data.sql [hunk 4]
```diff
@@ -4,3 +4,10 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon','2005-06-12',1,TRUE,45);
```
Comment (arslanbisharat) [inline data.sql]: - [ ] Please check the `weight_kg` is -45.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
### Hi @TesheMaximillan, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**